src/custom/priceApi/matcha-0x/index.ts

```diff
@@ -0,0 +1,137 @@
+import { OrderKind } from '@gnosis.pm/gp-v2-contracts'
+
+import { NetworkID } from 'paraswap'
+import { SupportedChainId as ChainId } from 'constants/chains'
+import { getTokensFromMarket } from 'utils/misc'
+import { getValidParams, PriceInformation, PriceQuoteParams } from 'priceApi/price'
+
+// https://github.com/0xProject/0x-api/blob/8c4cc7bb8d4fa06a220b7dfd5784361c05daa92a/src/types.ts#L229
+interface GetSwapQuoteResponseLiquiditySource {
+  name: string
+  proportion: string
+  intermediateToken?: string
+  hops?: string[]
+}
+
+// https://github.com/0xProject/0x-api/blob/8c4cc7bb8d4fa06a220b7dfd5784361c05daa92a/src/types.ts#L229
+interface MatchaBaseQuote {
+  chainId: ChainId
+  price: string
+  buyAmount: string
+  sellAmount: string
+  sources: GetSwapQuoteResponseLiquiditySource[]
+  gasPrice: string
+  estimatedGas: string
+  sellTokenToEthRate: string
+  buyTokenToEthRate: string
+  protocolFee: string
+  minimumProtocolFee: string
+  allowanceTarget?: string
+}
+
+// https://github.com/0xProject/0x-api/blob/8c4cc7bb8d4fa06a220b7dfd5784361c05daa92a/src/types.ts#L229
+export interface MatchaPriceQuote extends MatchaBaseQuote {
+  sellTokenAddress: string
+  buyTokenAddress: string
+  value: string
+  gas: string
+}
+
+function getMatchaChainId(chainId: ChainId): NetworkID | null {
+  switch (chainId) {
+    // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+    // See https://0x.org/docs/api#introduction
+    case ChainId.MAINNET:
+    case ChainId.ROPSTEN:
+      return chainId
+
+    default:
+      // Unsupported network
+      return null
+  }
```

Comment:
I'd say either have all of them (ropsten, polygon, BSC) or none of them, since I don't believe our contracts are on ropsten.
Or are they?